### PR TITLE
initial Fastlane structure

### PR DESCRIPTION
**Pull request by IzzySoft**
This PR provides initial Fastlane structures, mostly intended for your open [F-Droid inclusion](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/10312) but also useful for deployment to other places, like Google Play. This enables you to define how your app is presented, by keeping the included details the way you like them, optionally building on this to enhance (be welcome to my [Fastlane Cheat Sheet](https://gitlab.com/snippets/1895688) for that).

Screenshots, featureGraphic and icon are taken from your current Play Store listing. Of course feel free to replace anything you think needs replacement (like e.g. the first two screenshots, which seen to belong to `fastlane/metadata/ios` :wink:). Once this PR is merged, the next tag would make the structures available for F-Droid.

Should there be any questions, don't hesitate asking them.

### CODE REVIEW

**Comment by laurent22** [comment]
Thanks for the PR @IzzySoft. Just a few comments:

- Since it's only for mobile, it should be under packages/app-mobile

- All images should be pulled from their source and copied there using the generate-images.ts script. This is to ensure we don't have plenty of images to update everywhere when we change one of them. If you need help to get it working please let me know.

**Comment by IzzySoft** [comment]
> Since it's only for mobile, it should be under packages/app-mobile

Nope. It won't be found there. The `fastlane` directory **MUST** reside in the repo's root, sorry; this is a technical requirement.

> All images should be pulled from their source…

As I wrote: be welcome to build on this. If you wish I can remove the graphics from this PR so they can be dealt with separately; the "bare minimum" needed by F-Droid would be en-US summary & description.

> Please remove featureGraphic.png, because it's no longer current.

OK, will do. Shall I remove the screenshots and the icon as well?

**Comment by laurent22** [comment]
Hmm, ok but it's a monorepo and I don't want to add yet another directory at the root. I've tried before to build the app using fastlane and if we could do that it would be worth it, but just for a description I'd prefer we don't add this for now.

**Comment by Daeraxa** [comment]
Is there a reason it couldn't go in the joplin-android repo (or its own new one?) - i.e. does it have to derive from the main repo or can it just update when the build is run?

**Comment by IzzySoft** [comment]
> Is there a reason it couldn't go in the joplin-android repo

This was the repo referenced from the MR at F-Droid: `Repo: https://github.com/laurent22/joplin.git`. And the recipe went through just fine, resulting in an APK.

> or its own new one?

Yes, it must be in the same repo F-Droid builds from. I'm not sure, but theoretically it *may* be possible to put it into its own repo with the top-level `fastlane` directory skipped (i.e. `metadata/` at its top-level), and then pull that as git submodule named "fastlane" in the build or pre-build process (so it ends up as `fastlane/metadata/*` again). Another alternative would be using [Triple-T](https://gitlab.com/snippets/1901490) instead (which would go inside the build tree), but I'm not really familiar with that.

Third variant is as the MR at F-Droid is set up now, with summary & description inside the YAML. But then there'd be no graphics at all, which doesn't look that inviting. Is that directory really that disturbing?

Up to you of course – I just tried to help out making the listing as attractive as possible :wink:

**Comment by Daeraxa** [comment]
For reference the quotes above were me but I realised I was speaking from ignorance so deleted the comment about 4 femtoseconds after I made it to think about it a bit more rationally to actually look at the whole f-droid thing properly and understand it.

**Comment by IzzySoft** [comment]
@Daeraxa I wondered where it went to and then quoted from the notification mail, sorry :rofl:

**Comment by IzzySoft** [comment]
@laurent22 so is your "No" final (none of the variants viable to you), and you prefer living with the disadvantages described? Fastlane isn't *mandatory* for F-Droid, it's just highly recommended (for the reasons listed).

**Comment by laurent22** [comment]
> so is your "No" final (none of the variants viable to you), and you prefer living with the disadvantages described? Fastlane isn't _mandatory_ for F-Droid, it's just highly recommended (for the reasons listed).

Right, sorry I prefer not to add this at this point. I appreciate the pull request but we currently don't have a use for it in the monorepo itself. If we ever use Fastlane to build the mobile apps we'll add it at that time.

**Comment by yashpalgoyal1304** [comment]
> it's a monorepo and I don't want to add yet another directory at the root

i kinda agree to that, too many directories at the root become kinda hard to follow....
